calico/felix/config.py

```diff
@@ -463,7 +469,51 @@ def _validate_cfg(self, final=True):
         if self.ETCD_CA_FILE == "none":
             self.ETCD_CA_FILE = None
 
-        if self.ETCD_SCHEME == "https":
+        # Determine which etcd scheme and addresses will be used.
+        addrs = []
```

Comment:
How about pushing this parsing into a utility method.  Feels like a pretty clean pure function that's make it easy to throw lots of UT example at.